runtime/jvmti/jvmtiCapability.c

```diff
@@ -372,7 +372,7 @@ jvmtiAddCapabilities(jvmtiEnv* env,
 		}
 
 #if JAVA_SPEC_VERSION >= 11
-		if (newCapabilities.can_generate_sampled_object_alloc_events) {
+		if (capabilities_ptr->can_generate_sampled_object_alloc_events) {
```

Comment:
I think `newCapabilities` was correct.  The loop above does two things:
* `/* Mask out any currently-owned capabilities */`
* `/* Make sure all of the requested capabilities are available */`

If this environment already has the `can_generate_sampled_object_alloc_events`, then it shouldn't be an error for it to request it again.  May need to confirm with the spec

Reply:
Thanks @DanHeidinga , that's right. `newCapabilities` masked out currently-owned capabilities hence the code within this `if (true)` conditional block will be executed only when `can_generate_sampled_object_alloc_events` is requested and not enabled before.
The spec doesn't explicitly state if it is ok to request same capabilities more than once however `RI` does allow such behavior, so does `OpenJ9`.

Updated the PR to remove this modification, please review again.